### PR TITLE
Bump rousette to latest GIT hash

### PR DESCRIPTION
**Pull request by mattiaswal**
Since we are using an old GCC we need to add a new requirement for rousette 'date', this is included in GCC 14+.

## Checklist

Tick *relevant* boxes, this PR is-a or has-a:

- [ ] Bugfix
  - [ ] Regression tests
  - [ ] ChangeLog updates (for next release)
- [ ] Feature
  - [ ] YANG model change => revision updated?
  - [ ] Regression tests added?
  - [ ] ChangeLog updates (for next release)
  - [ ] Documentation added?
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (please detail in commit messages)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe):

### CODE REVIEW

File: package/Config.in [hunk 1]
```diff
@@ -32,4 +32,5 @@ source "$BR2_EXTERNAL_INFIX_PATH/package/libyang-cpp/Config.in"
 source "$BR2_EXTERNAL_INFIX_PATH/package/sysrepo-cpp/Config.in"
 source "$BR2_EXTERNAL_INFIX_PATH/package/rousette/Config.in"
 source "$BR2_EXTERNAL_INFIX_PATH/package/nghttp2-asio/Config.in"
+source "$BR2_EXTERNAL_INFIX_PATH/package/date/Config.in
```
Comment (troglobit) [inline package/Config.in:35]: I suggest we use consistent naming, as far as possible, and call it `date-cpp`.

File: package/date/Config.in [hunk 2]
```diff
@@ -0,0 +1,5 @@
+config BR2_PACKAGE_DATE
+	bool "date"
```
Comment (troglobit) [inline package/date/Config.in:2]: Same here, `BR2_PACKAGE_DATE_CPP`, and "date-cpp".

File: package/date/Config.in [hunk 3]
```diff
@@ -0,0 +1,5 @@
...
+	help
+	  A date and time library based on the C++11/14/17 <chrono> header
+	  https://github.com/HowardHinnant/date
```
Comment (troglobit) [inline package/date/Config.in:5]: Need an empty line before the URL.

**Review by troglobit — CHANGES_REQUESTED** [review]
Some minor suggestions.  Amazingly quick turnaround on this one! :rocket::sparkles:

File: package/Config.in (hunk 1 above)
Comment (mattiaswal) [reply package/Config.in:35]: I was thinking that, but the project is named "date"

Comment (troglobit) [reply package/Config.in:35]: Yeah I noticed, but plain date is way too generic. We don't want any inadvertent collisions with buildroot, or other unintended misunderstandings.

**Review by troglobit — APPROVED** [review]
Nice work! :boom:

**Pull request merged**